src/chunk.c

```diff
@@ -1461,6 +1513,15 @@ ts_chunk_create_for_point(const Hypertable *ht, const Point *p, bool *found, con
 	 * conflicts with itself. The lock needs to be held until transaction end.
 	 */
 	LockRelationOid(ht->main_table_relid, ShareUpdateExclusiveLock);
+	/* If we have a compressed hypertable set, lock up the compressed hypertable
+	 * so we can create the compressed chunk too
+	 */
+	if (ht->fd.compressed_hypertable_id != 0)
+	{
+		compress_ht = ts_hypertable_get_by_id(ht->fd.compressed_hypertable_id);
+		Assert(compress_ht);
+		LockRelationOid(compress_ht->main_table_relid, ShareUpdateExclusiveLock);
+	}
```

Comment:
Looking further down the call chain, I get the following (`chunk_create_from_point_after_lock@1574` -> `chunk_create_from_hypercube_after_lock@1402`):

```
(gdb) l
1216             * to reduce locking contention since we already use heavy locks to attach
1217             * chunks to the hypertable.
1218             */
1219            if (ht->fd.compressed_hypertable_id != 0)
1220            {
1221                    chunk->fd.compressed_chunk_id =
1222                            create_compressed_chunk(ht->fd.compressed_hypertable_id, schema_name);
1223            }
1224
1225            chunk_add_constraints(chunk);
(gdb) p chunk->fd.compressed_chunk_id 
$6 = 0
```

So it seems like it is not set.

Reply:
At this point, we are creating a new chunk so we need to create the compressed chunk for it as well since it did not exist when enabling compression on the hypertable.

Reply:
Hmm, I guess this if for new chunks that didn't exist when compression was enabled.

I do wonder, however, if we can delay grabbing the ShareUpdateExclusiveLock on the compressed hypertable until we know we actually needed the lock on the main hypertable. See check below for concurrent creation by other backend. In other words, if someone else created the chunk before we grabbed the lock, we need not take the lock on the compressed hypertable at all.